enumeratum-core/src/main/scala/enumeratum/EnumEntry.scala

```diff
@@ -61,4 +61,17 @@ object EnumEntry {
     abstract override def entryName: String = super.entryName.toLowerCase
   }
 
+  /**
+   * Helper implicit for more readable checking.
+   */
+  implicit class ComparableEnum[A <: EnumEntry](enum: A) {
```

Comment:
Yes, it is clear now. Using value class is a good idea, indeed. I modified the code accordingly.